### ISSUE TITLE
Asking for Controller Code of AutoAugment

### ISSUE THREAD

**Issue by WolfNiu** [tensorflow/models#5741]
Could someone let me know whether the controller part of the code is released? By inspecting the autoaugment folder, I only found an already trained policy. Thank you.

**Comment by JaspreetSambee** [comment]
I am also looking for the same. btw, did you try to train any of the models from the repo with the trained AutoAugment policies? I tried and the training is heck slow, probably due to the feed_dict method being used and that each image augmentation might be taking too long. Any cues here would be helpful.

**Comment by WolfNiu** [comment]
> I am also looking for the same. btw, did you try to train any of the models from the repo with the trained AutoAugment policies? I tried and the training is heck slow, probably due to the feed_dict method being used and that each image augmentation might be taking too long. Any cues here would be helpful.

Did you specify "use_cpu=0"? If it is 1 then the model will train on your cpu.

**Comment by JaspreetSambee** [comment]
> > I am also looking for the same. btw, did you try to train any of the models from the repo with the trained AutoAugment policies? I tried and the training is heck slow, probably due to the feed_dict method being used and that each image augmentation might be taking too long. Any cues here would be helpful.
> 
> Did you specify "use_cpu=0"? If it is 1 then the model will train on your cpu.

Well I did that but with every step of training (i.e session.run() call every step), the time to run one session.run() is increasing every step. So the training starts well but then it just becomes really slow as the time just increases with every step.

**Comment by WolfNiu** [comment]
I see. This could be environment specific since I don't have that problem. The issue I'm having is that training accuracy shows normally, while valid/test accuracies all show up as 0.0

**Comment by JaspreetSambee** [comment]
I was able to fix the problem of the increasing training time as i figured that the fetching of LR every step was somehow causing an overhead to the graph size, dont know how though. The training is going well and i can see the valid/test accuracies computed properly. Yet to see how well the model does since i am trying to train ResNet50 on Cifar10 using AutoAugment.

**Comment by WolfNiu** [comment]
> I was able to fix the problem of the increasing training time as i figured that the fetching of LR every step was somehow causing an overhead to the graph size, dont know how though. The training is going well and i can see the valid/test accuracies computed properly. Yet to see how well the model does since i am trying to train ResNet50 on Cifar10 using AutoAugment.

Glad that your issue got resolved. The info you mentioned is also very important to me. Thanks.

**Comment by JaspreetSambee** [comment]
I tried training the ResNet50 with Cifar10, the original TF repo achieves ~93% validation accuracy, but with the AutoAugment policies acting as the augmentation, it can reach only 91.5% validation accuracy after all the hyperparameter tuning. I was expecting the AutoAugment policy to perform better generally and get even higher accuracy than the traditional methods but couldn't in this case. Maybe I am missing something here.

**Comment by sherdencooper** [comment]
> I see. This could be environment specific since I don't have that problem. The issue I'm having is that training accuracy shows normally, while valid/test accuracies all show up as 0.0

Hello, i also had that problem. Have you ever solved the problem? Any advice is helpful, thanks a lot.

**Comment by ZyuZyuZyu** [comment]
@JaspreetSambee May I ask you some questions? How were you able to train a Resnet50 with autoaugment, Is there a controller or you have coded one?Thanks!

**Comment by JaspreetSambee** [comment]
> @JaspreetSambee May I ask you some questions? How were you able to train a Resnet50 with autoaugment, Is there a controller or you have coded one?Thanks!

Hey, no I did not use the Controller part. I just used the final Augmentation policies mentioned in the AutoAugment paper for Cifar10 and tried to apply that policy on to my Resnet50 code, in the hope that this would increase the accuracy of the baseline ResNet50 accuracy on Cifar10, but unfortunately I could not get higher accuracy with that. The baseline accuracy(with basic augmentation) was ~93% but with autoaugment, it could only reach ~91.5%.

**Comment by TianyuanYu** [comment]
@WolfNiu @sherdencooper 
The val accuracy is 0 because cifar does not have validation set. If replace `val` with `test`, the test result will be given after and before each epoch.

@JaspreetSambee 
I think the controller is the key. It is not fair compared if you just delete that part.

I find it is extremely slow. With the shake_drop model, gpu-mode experiment takes 15 mins each epoch and cpu-mode takes much longer. Can I ask the training speed in your experiments?

**Comment by fnzhan** [comment]
Hi, have you found the code of the search algorithm to find the policies?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by amahendrakar** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
